### ISSUE TITLE
StreamingTensorNormalizer variance estimate is inconsistent with the docstring.

### ISSUE THREAD

**Issue by johnamcleod** [tensorflow/agents#530]
My understanding of the `StreamingTensorNormalizer` is that all of the data should be passed in batches to the `update` method, which will update the internal state so that when the latest batch of data is passed to `normalize` it is normalized with respect to the mean and the (biased) estimate of the variance of all of the previously observed data. At least, this is how I interpret the docstrings. 

If that is correct, then I do not believe that the current implementation for the variance is correct. I would expect that, after a single batch of data, the variance estimate (returned from `_get_mean_var_estimates`) should be the (biased) estimate of the variance of the single batch. That does not appear to be true.

For example:

```
    normalizer = tensor_normalizer.StreamingTensorNormalizer(
        tensor_spec.TensorSpec([], tf.float32), scope='normalize_advantages'
    )
    tensor = tf.constant([1., 2.])

    normalizer.update(tensor, outer_dims=(0,))
    mean, var = normalizer._get_mean_var_estimates()
```

The mean value matches `tf.reduce_mean(tensor)`. I would expect that the variance should match `tfp.stats.variance(tensor)` which is also a biased estimate of the variance of the values in a tensor. However, here `var = 2.5` and `tfp.stats.variance(tensor) = 0.25`.

**Comment by sguada** [comment]
Can you create a test to verify if the behavior is correct or not?

**Comment by johnamcleod** [comment]
I have raised a PR with a test which (I believe) shows that the behaviour is not correct: #543

**Comment by johnamcleod** [comment]
Here the docstring of the `StreamingTensorNormalizer` says it uses the full history of tensor values:
https://github.com/tensorflow/agents/blob/master/tf_agents/utils/tensor_normalizer.py#L244

Here the docstring in the `_update_ops` method says it maintains a biased estimate of the variance:
https://github.com/tensorflow/agents/blob/master/tf_agents/utils/tensor_normalizer.py#L274

I have written a test which checks that the variance estimate after updating with a single data set is equal to the biased estimate of the variance of that data set. This compares the output to https://www.tensorflow.org/probability/api_docs/python/tfp/stats/variance which also claims to compute a biased estimate of the variance.
